core/src/main/java/io/temporal/samples/earlyreturn/TransactionActivitiesImpl.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.earlyreturn;
+
+import io.temporal.failure.ApplicationFailure;
+
+public class TransactionActivitiesImpl implements TransactionActivities {
+
+  @Override
+  public Transaction initTransaction(TransactionRequest txRequest) {
+    System.out.println("Initializing transaction");
+    sleep(500);
+    if (txRequest.getAmount() <= 0) {
+      System.out.println("Invalid amount: " + txRequest.getAmount());
+      throw ApplicationFailure.newNonRetryableFailure(
+          "Non-retryable Activity Failure: Invalid Amount", "InvalidAmount");
+    }
+    // Simulate transaction ID generation
+    String txId = "TXID" + String.format("%010d", (long) (Math.random() * 1_000_000_0000L));
+    Transaction tx =
+        new Transaction(
+            txId,
+            txRequest.getSourceAccount(),
+            txRequest.getTargetAccount(),
+            txRequest.getAmount());
+
+    sleep(500);
```

Comment:
nit: is the second sleep necessary?

Reply:
It's intentional. I wanted to give the impression of a failed initialization taking less long than a successful one (since the failed one throws an exception 'early').

Since the sleeps are for artistic effect, this is me artistically expressing myself ;) Let me know if you disagree with this approach and I'll do whatever you want

Reply:
👍 makes sense!